### ISSUE TITLE
Developer Guide review

### ISSUE THREAD

**Issue by kohnh** [AY2223S2-CS2113-W12-3/tp#115]
## Some of the errors are repeated, take note to double check for all your diagrams.
Some general notes:
- Take note of call to static methods (if any)
![image](https://user-images.githubusercontent.com/64647635/228826025-462a9307-a02d-4c1c-b195-e5a7824160c5.png)
- Try to include class diagrams and NFR. Those templates sections should be removed if nothing is added.

### 1.
![image](https://user-images.githubusercontent.com/64647635/228820223-d57adb96-496f-4acc-b46a-44eb9c0dfe2a.png)
<br>
### 2.
![image](https://user-images.githubusercontent.com/64647635/228822603-649c23a0-6813-44ff-b04c-cd5bee98ca60.png)
A combined alt table might make more sense since it is an `if...else if...`
![image](https://user-images.githubusercontent.com/64647635/228821218-ae4e2e2b-6e26-4549-aa7a-094d258ba0ed.png)
`sd` more so refers to a reference frame that have been omitted from somewhere but contains important information that you feel should be included thus you show in an alternate sequence diagram. But in this case I did not find a main sequence diagram (if there is maybe you can include that before this so there will be a better flow) so maybe you can omit the `sd` wrapping block.
<br>
### 3.
![image](https://user-images.githubusercontent.com/64647635/228823036-6822a4b0-990c-4acc-99b4-1f0010466d97.png)
In object diagrams entities should have a colon and be underlined.
The visibility should also be according to the module website.
In this case, inventory does not seem to be a primitive type maybe can be abstracted to the `Inventory` object and putting `inventory = inventory` does not provide much value, you can try giving an example case where you provide a random valid item.

### 4.
![image](https://user-images.githubusercontent.com/64647635/228825553-753d9977-1a6e-404b-8a89-99ad3e7ff97c.png)

### 5.
![image](https://user-images.githubusercontent.com/64647635/228842579-cbcb359c-da61-4cb1-8a78-902269b73e09.png)
There is no need to show the specific method calls then afterwards using `ref` to explain a specific part. It would be easier to explain in words without boring the reader with unnecessary details. Additionally, deep nesting may reduce the readability.
Additionally, this diagram on the right seems to be a reference frame of the diagram on the left. In such a case you should wrap your diagram on the right with a `ref` frame and add the appropriate naming so that it would be easier for readers to link the two together.
![image](https://user-images.githubusercontent.com/64647635/228842792-0561ee83-523e-43db-b775-17800a310507.png)
Refer to this diagram from the module website for how to show reference frames.
![image](https://user-images.githubusercontent.com/64647635/228841230-36b7b8bb-79eb-463a-a343-18d20047a75a.png)
When you mention create a new object, you should show a constructor in your sequence diagram.

### 6.
![image](https://user-images.githubusercontent.com/64647635/228843295-363991e2-186b-4c40-8c70-5332b0b86a5b.png)
Information seems to be repeated

### 7.
![image](https://user-images.githubusercontent.com/64647635/228843993-cab1fd4e-f7f7-4179-8aa4-c5718cf6b516.png)
Where the control flow is returned is important. In the first two alt cases, the control is not returned and the object is not deleted.

### 8.
![image](https://user-images.githubusercontent.com/64647635/228845409-5c3ba367-98ae-4f8e-b65a-1cd18d2e5abe.png)
Recall what we did in tutorial 10, the constructor block should be separated from the activation block since in this case there is a call of `run()` from user.

### 9.
![image](https://user-images.githubusercontent.com/64647635/228850287-426d569e-c5a2-43f3-9274-1aad64386a6d.png)
There should not be empty `alt` blocks

### 10.
![image](https://user-images.githubusercontent.com/64647635/228850685-19e92140-c81a-4284-b6e2-d2a4b6bf73ec.png)
You can combine both table so that it looks neater

### 11.
![image](https://user-images.githubusercontent.com/64647635/228851362-c773a8d7-766d-447c-9d9d-fe12a501b3c1.png)
Details can be generalized if not it will be too wordy.

### 12.
![image](https://user-images.githubusercontent.com/64647635/228852133-a8eca2b1-1128-40e8-9e4c-01dbc8e6c487.png)
Take note of spelling mistakes

**Comment by firwer** [comment]
Thanks for the feedback @kohnh
